File: 12-BIT-MANIPULATION/PHP/counting-bits.php
<?php

// https://leetcode.com/problems/counting-bits/

class Solution
{
    function countBits(int $n): array
    {
        $bits = array_fill(0, $n+1, 0);
        for ($i = 1; $i <= $n; $i++) {
            $bits[$i] = $bits[$i >> 1] + ($i & 1);
        }

        return $bits;
    }
}

File: 12-BIT-MANIPULATION/Python/counting-bits.py
from typing import List

# https://leetcode.com/problems/counting-bits/

class Solution:
    def countBits(self, n: int) -> List[int]:
        bits = [0] * (n + 1)
        for i in range(1, n+1):
            bits[i] = bits[i >> 1] + (i & 1)
        return bits